### DISCUSSION TITLE
HTML in a template variable

### DISCUSSION THREAD

**Discussion by poggingfish** [vlang/v#19186] [Questions and Answers] [answered]
I am using vweb to create a website. the variable `post.first().rating_message` contains HTML in a string.
How do i parse this?

```html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>${title}</title>
    @css '/templates/page/dark.css'
</head>
<body>
    <h1>Review of ${post.first().name}</h1>
    <p>${post.first().desc}</p>
    <p>Posted at ${post.first().time} by ${admin}</p>
    <h2>This thing got ${post.first().rating} out of 10 stars!</h2>
    <hr></hr>
    $post.first().rating_message
    @include 'footer'
</body>
</html>
```

See in the image below that the HTML is not being rendered.

![image](https://github.com/vlang/v/assets/38115495/06e3044b-26f1-4945-863b-ce3617376841)

**Comment by poggingfish** [comment, marked as answer]
Okay, after doing some trickery I found a solution.
Instead of using `return $vweb.html()` like you usually do, I instead did `return app.html($tmpl("templates/page/view.html"))`.

Maybe there should be a feature implemented so there is a way to do this with `$vweb.html`?

**Comment by medvednikov** [comment]
In your HTML template you can do

```@app.html(post.first().rating_message)```

**Reply by poggingfish** [reply]
Thats good to know! Thank you :)
